CRM/ManualDirectDebit/Mail/DataCollector/Base.php

```diff
@@ -203,6 +216,37 @@ private function collectRecurringContributionData() {
     ];
   }
 
+  /**
+   * Builds a HTML table for recurring contribution installments
+   *
+   * @param $recurringContributionPlan
+   * @return string
+   */
+  private function buildRecuringContributionTable($recurringContributionPlan) {
+    $html = '<table>'
```

Comment:
I think you should try moving the html code to a template and use smarty to process it and return the resulting string. Something like:

```
private function buildRecuringContributionTable($recurringContributionPlan) {
  $smarty = CRM_Core_Smarty::singleton();
  $smarty->assign('installments', $recurringContributionPlan);

  return $smarty->fetch('CRM/ManualDirectDebit/MessageTemplate/InstallmentList.tpl');
}
```